Epsilon.Host.Frontend/src/components/Competence/CompetenceProfileCell.vue

```diff
@@ -1,16 +1,47 @@
+<template>
+    <td v-if="result == null || count == 0"></td>
+    <td v-else :style="{ background: '#' + result?.value?.hexColor }">
+        {{ count }}
+    </td>
+</template>
+
 <script setup lang="ts">
-import { defineProps } from "vue"
-import { LearningDomainOutcome } from "@/api"
+import { computed, defineProps } from "vue"
+import {
+    LearningDomainOutcome,
+    LearningDomainSubmission,
+    LearningDomainType,
+} from "@/api"
 
-defineProps<{
+const props = defineProps<{
+    submissions: LearningDomainSubmission[]
     result: LearningDomainOutcome | null
+    row: LearningDomainType
+    col: LearningDomainType
 }>()
+/*
+Count all the outcomes that have been graded to mastery within a row and column 
+ */
+const count = computed(() => {
+    return props.submissions
+        .map(
+            (s) =>
+                s.results?.filter(
+                    (r) =>
+                        r.outcome?.row?.id == props.row.id &&
+                        r.outcome?.column?.id == props.col.id &&
+                        (r.grade as number) >= 3
```

Comment:
Maybe just check if results is null beforehand so you don't need all of these nullary operators `?`. Outcomes never return null from the results, and only the column of the outcome may be null.